test/jit/test_cuda.py

```diff
@@ -0,0 +1,410 @@
+            e_sync.record(torch.cuda.current_stream(device_index))
+            e_sync.wait(s1)
+            with torch.jit.cuda.stream(s1):
+                tensor3 = torch.rand(1000000000, 1000000000, device = "cuda")
+                tensor4 = torch.mm(tensor3, tensor3).cuda()
+            s1.synchronize()
+            e_tok.record(torch.cuda.current_stream(device_index))
+            e_tok.synchronize()
+            s0.synchronize()
+
+            if not s0.query() or not s1.query() or not e_sync.query():
+                return -1.0
+
+            # not necessary to check e_tik and e_tok, as elapsed_time would throw
+            # exception if otherwise.
+            return e_tik.elapsed_time(e_tok)
+        self.assertGreater(test_event_wait(), 0)
+
+        @unittest.skipIf(not TEST_MULTIGPU, "detected only one GPU")
+        @unittest.skipIf(not TEST_LARGE_TENSOR, "not enough memory")
+        @torch.jit.script
+        def test_events_wait():
+            d1 = torch.device('cuda:1')
+
+            with torch.jit.cuda.device(d1):
+                s0 = torch.cuda.current_stream(1)
+                tensor1 = torch.rand(1000000000, 1000000000, device = "cuda")
+                tensor2 = torch.mm(tensor1, tensor1).to("cuda")
+                e0 = torch.jit.cuda.Event(False, False, False)
+                s0.record_event(e0)
+
+            s1 = torch.cuda.current_stream(0)
+            s1.wait_event(e0)
+            s1.synchronize()
+
+            return e0.query() and s0.query() and s1.query()
+        self.assertTrue(test_events_wait())
+
+    @skipIfRocm
+    def test_save_load(self):
+        class Model(torch.nn.Module):
+            def forward(self):
+                device_index = torch.cuda.current_device()
+                s = torch.jit.cuda.Stream(device_index, 0)
+                a = torch.rand(3, 4, device = "cuda")
+                b = torch.rand(3, 4, device = "cuda")
+
+                with torch.jit.cuda.stream(s):
+                    is_stream_s = torch.cuda.current_stream(s.device_index()).id() == s.id()
+                    c = torch.cat((a, b), 0).cuda()
+                s.synchronize()
+                return is_stream_s, a, b, c
+
+        model = Model()
+
+        # Script the model and save
+        script_model = torch.jit.script(model)
+        is_stream_s, a, b, c = script_model()
+        # Verify if the output is correct
+        self.assertTrue(is_stream_s)
+        self.assertEqual(torch.cat((a, b), 0), c)
+
+        saved_model = BytesIO()
+        torch.jit.save(script_model, saved_model)
+
+        # copy the data in the saved_model so we can restore it later. This
+        # is because py2 and py3 have different semantics with zipfile
+        saved_model_copy = saved_model.getvalue()
+
+        # Load the model from the copy into a buffer and compare
+        # the output with the saved model
+        buffer = BytesIO(saved_model_copy)
+        load_model = torch.jit.load(buffer)
```

Comment:
We have a helper function called `getExportImportCopy` that you might be able to use here to save some lines of code.

Reply:
Cool, thanks. This works :)